### PR TITLE
refactor(Scripts/Commands): Update cs_quest to the new model

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:
-  Update cs_quest to the new parsing model
-  Allow all commands to be used from console and to target offline players

## Issues Addressed:

- Closes 

## Tests Performed:

- Tested in-game.
- 

## How to Test the Changes:

1. quest add 102 $charname
2. quest complete 102 $charname
3. Log in, check if the quest is complete

...
Test using the commands when characters are online/offline, from console and in-game.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
@Nyeriah ping me after done this

**Comment by Nyeriah** [comment]
@Winfidonarleyan I'm done with this

File: src/server/game/Quests/QuestDef.cpp [hunk 1]
```diff
@@ -184,48 +184,43 @@ void Quest::LoadQuestTemplateAddon(Field* fields)
...
-        else if (xp <= 500)
-            xp = 10 * ((xp + 5) / 10);
-        else if (xp <= 1000)
-            xp = 25 * ((xp + 12) / 25);
-        else
-            xp = 50 * ((xp + 25) / 50);
+    int32 quest_level = (Level == -1 ? playerLevel : Level);
+    const QuestXPEntry* xpentry = sQuestXPStore.LookupEntry(quest_level);
+    if (!xpentry)
+        return 0;
```
Comment (Nefertumm) [inline src/server/game/Quests/QuestDef.cpp:192]: brackets

File: src/server/game/Quests/QuestDef.cpp [hunk 2]
```diff
@@ -184,48 +184,43 @@ void Quest::LoadQuestTemplateAddon(Field* fields)
...
+
+    uint32 xp = diffFactor * xpentry->Exp[RewardXPDifficulty] / 10;
+    if (xp <= 100)
+        xp = 5 * ((xp + 2) / 5);
+    else if (xp <= 500)
+        xp = 10 * ((xp + 5) / 10);
+    else if (xp <= 1000)
+        xp = 25 * ((xp + 12) / 25);
+    else
+        xp = 50 * ((xp + 25) / 50);
```
Comment (Nefertumm) [inline src/server/game/Quests/QuestDef.cpp:208]: brackets here

**Review by Nefertumm — APPROVED** [review]

File: src/server/game/Quests/QuestDef.cpp (hunk 2 above)
Comment (Nyeriah) [reply src/server/game/Quests/QuestDef.cpp:208]: I had the misfortune of editing 10-years-old code :(

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by Nyeriah** [comment]
Skärmbild 2021-11-22 123050

File: src/server/game/Miscellaneous/Language.h [hunk 3]
```diff
@@ -1225,6 +1226,10 @@ enum AcoreStrings
     LANG_COMMAND_CACHE_REFRESH          = 5065,
     LANG_COMMAND_CACHE_NOT_FOUND        = 5066,
 
+    LANG_COMMAND_QUEST_ADD              = 5067,
+    LANG_COMMAND_QUEST_NOT_FOUND_IN_LOG = 5068,
+    LANG_COMMAND_QUEST_NOT_COMPLETE     = 5069,
+
     // Room for more strings    5063-9999
```
Comment (Nefertumm) [inline src/server/game/Miscellaneous/Language.h:1233]: Update this one too 😃

**Comment by Nefertumm** [comment]
Neat!

**Pull request merged**